### PR TITLE
[csl-2811]: add answer option text to answerInputs

### PR DESCRIPTION
**Pull request by HHHindawy**
**This PR adds more details to the `answerInputs` array**

_Notice `answerInputs[5].value` & `answerInputs[6].value` -> Used to be an array of `string` now it's an array of objects consisting of `id` & `value`:_

**Before:**

image

**After:**

image

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export interface QuizReturnState {
 
 export type AnswerInput = {
   type: InputQuestionsTypes;
-  value: string | string[];
+  value: string | Partial<QuestionOption>[];
```
Comment (mocca102) [inline src/types.ts:99]: Will this ever have any other props other than id and value? If not can we use Omit<> instead and omit the unused props?

File: src/components/CioQuiz/quizLocalReducer.ts [hunk 2]
```diff
@@ -76,10 +76,12 @@ export default function quizLocalReducer(
...
           break;
         case QuestionTypes.SingleSelect:
-          newAnswers.push(currentAnswerInput.value as string[]);
-          break;
         case QuestionTypes.MultipleSelect:
-          newAnswers.push(currentAnswerInput.value as string[]);
+          newAnswers.push(
+            (typeof currentAnswerInput.value === 'string'
+              ? [currentAnswerInput.value]
+              : currentAnswerInput.value.map((answer) => answer.id)) as string[]
```
Comment (mocca102) [inline src/components/CioQuiz/quizLocalReducer.ts:83]: I think if we did the Omit thing id will be required and you won't have to use `as string[]` here

**Review by mocca102 — CHANGES_REQUESTED** [review]
This is looking good just left a couple of comments for typescript

File: src/types.ts (hunk 1 above)
Comment (HHHindawy) [reply src/types.ts:99]: True 👍

**Review by mocca102 — APPROVED** [review]

**Review by jjl014 — COMMENTED** [review]
Nice work, this is looking great to me! 🚀 

I would love to get some thoughts on the comment I had below. If it sounds good, I think it would be an easy change to incorporate into this PR (but lmk otherwise). 😄

File: src/components/CioQuiz/quizLocalReducer.ts [hunk 3]
```diff
@@ -76,10 +76,12 @@ export default function quizLocalReducer(
...
         case QuestionTypes.SingleSelect:
-          newAnswers.push(currentAnswerInput.value as string[]);
-          break;
         case QuestionTypes.MultipleSelect:
-          newAnswers.push(currentAnswerInput.value as string[]);
+          newAnswers.push(
+            (currentAnswerInput.value as Omit<QuestionOption, 'attribute' | 'images'>[]).map(
+              (answer) => answer.id
+            )
+          );
```
[the remark this answers is not in the document]
Comment (jjl014) [reply src/components/CioQuiz/quizLocalReducer.ts:84]: I just realized the Skip code wasn't added by us. @HHHindawy, feel free to ignore this.

This PR looks great to me and I'll merge/ship it sometime today!

**Pull request merged**